### PR TITLE
store/tikv:make KVTxn in tikv public

### PR DESCRIPTION
**Pull request by AndreMouche**
Signed-off-by: AndreMouche

### What problem does this PR solve?
This PR make the transaction in tikv public, so we needn't share the same Transaction or Storage interface with tidb/kv package
Part of https://github.com/pingcap/tidb/issues/22513

### Check List 
Tests 
- Unit test

### Release note 
- No release note

### CODE REVIEW

File: store/driver/txn/txn_driver.go [hunk 1]
```diff
@@ -0,0 +1,15 @@
+package txn
```
Comment (jackysp) [inline store/driver/txn/txn_driver.go:1]: Need a license.

**Review by disksing — APPROVED** [review]

**Review by andylokandy — APPROVED** [review]

**Comment by AndreMouche** [comment]
/merge

**Pull request merged**